core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -802,15 +817,24 @@ private[spark] class AppStatusListener(
     }
   }
 
-  override def onStageExecutorMetrics(executorMetrics: SparkListenerStageExecutorMetrics): Unit = {
+  override def onStageExecutorMetrics(event: SparkListenerStageExecutorMetrics): Unit = {
     val now = System.nanoTime()
-
+    Option(liveStages.get((event.stageId, event.stageAttemptId)))
+      .foreach { stage =>
+      if (stage.peakExecutorMetrics.compareAndUpdatePeakValues(event.executorMetrics)) {
+        update(stage, now)
+      }
+      val esummary = stage.executorSummary(event.execId)
+      if (esummary.peakExecutorMetrics.compareAndUpdatePeakValues(event.executorMetrics)) {
+        update(esummary, now)
+      }
+    }
     // check if there is a new peak value for any of the executor level memory metrics,
     // while reading from the log. SparkListenerStageExecutorMetrics are only processed
     // when reading logs.
```

Comment:
I need a little refresher here -- does this comment also apply to the new code you're adding above?

and why again does this only apply for when we're reading from the event logs (maybe this comment should be updated to point to whatever is happening for a live app)

Reply:
Yes, comments are still applicable, although for stage level. I'll move the comments around, and and add a pointer to onExecutorMetricsUpdate for the live app case.